### ISSUE TITLE
[BUG] dev dependencies not installed if marked as optional peer dependencies

### ISSUE THREAD

**Issue by lpinca** [npm/cli#2020]
### Current Behavior:

When a dependency is specified as dev dependency and as optional peer dependency, it is not installed.

### Expected Behavior:

I expect all dev dependencies to be installed.

### Steps To Reproduce:

```
luigi@ubuntu:~/ws$ node -v
v15.0.1
luigi@ubuntu:~/ws$ npm -v
7.0.3
luigi@ubuntu:~$ git clone https://github.com/websockets/ws.git
Cloning into 'ws'...
remote: Enumerating objects: 14410, done.
remote: Total 14410 (delta 0), reused 0 (delta 0), pack-reused 14410
Receiving objects: 100% (14410/14410), 10.68 MiB | 969.00 KiB/s, done.
Resolving deltas: 100% (11804/11804), done.
luigi@ubuntu:~$ cd ws
luigi@ubuntu:~/ws$ npm i
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142

added 352 packages, and audited 352 packages in 25s

33 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
luigi@ubuntu:~/ws$ npm ls utf-8-validate
ws@7.3.1 /home/luigi/ws
└── (empty)
```

### Environment:

- OS: Ubuntu 20.04
- Node: 15.0.1
- npm: 7.0.3

**Comment by ljharb** [comment]
Duplicate of #1998?

**Comment by lpinca** [comment]
It's not a duplicate. #1998 did not use optional peer dependencies and had `omit=peer` in `.npmrc`.

**Comment by lpinca** [comment]
Refs: https://github.com/nodejs/citgm/issues/836

**Comment by lpinca** [comment]
A possible hint from https://github.com/npm/cli/issues/2005#issuecomment-713884822

> What's happening here is that we're loading the peer edge ahead of the dev edge, and then skipping it. And since the peer edge is optional, we don't install it. For the root project at least, `dev` should take priority, I agree, so this is a bug.

It could be fixed by https://github.com/npm/arborist/pull/166.

**Comment by lpinca** [comment]
The issue was fixed in `npm@7.0.4`.
